openvdb_points/tools/AttributeArray.h

```diff
@@ -1658,56 +1702,59 @@ bool AttributeHandle<T, Strided, Interleaved>::isUniform() const
 
 // AttributeWriteHandle implementation
 
-template <typename T, bool Strided, bool Interleaved>
-typename AttributeWriteHandle<T, Strided, Interleaved>::Ptr
-AttributeWriteHandle<T, Strided, Interleaved>::create(AttributeArray& array)
+template <typename ValueType, typename CodecType, bool Strided, bool Interleaved>
+typename AttributeWriteHandle<ValueType, CodecType, Strided, Interleaved>::Ptr
+AttributeWriteHandle<ValueType, CodecType, Strided, Interleaved>::create(AttributeArray& array, const bool expand)
 {
-    return  typename AttributeWriteHandle<T, Strided, Interleaved>::Ptr(
-            new AttributeWriteHandle<T, Strided, Interleaved>(array));
+    return  typename AttributeWriteHandle<ValueType, CodecType, Strided, Interleaved>::Ptr(
+            new AttributeWriteHandle<ValueType, CodecType, Strided, Interleaved>(array, expand));
 }
 
-template <typename T, bool Strided, bool Interleaved>
-AttributeWriteHandle<T, Strided, Interleaved>::AttributeWriteHandle(AttributeArray& array)
-    : AttributeHandle<T, Strided, Interleaved>(array, /*preserveCompression = */ false) { }
+template <typename ValueType, typename CodecType, bool Strided, bool Interleaved>
+AttributeWriteHandle<ValueType, CodecType, Strided, Interleaved>::AttributeWriteHandle(AttributeArray& array, const bool expand)
+    : AttributeHandle<ValueType, CodecType, Strided, Interleaved>(array, /*preserveCompression = */ false)
+{
+    if (expand)     array.expand();
+}
 
-template <typename T, bool Strided, bool Interleaved>
-void AttributeWriteHandle<T, Strided, Interleaved>::set(Index n, const T& value)
+template <typename ValueType, typename CodecType, bool Strided, bool Interleaved>
+void AttributeWriteHandle<ValueType, CodecType, Strided, Interleaved>::set(Index n, const ValueType& value)
 {
-    this->mSetter(const_cast<AttributeArray*>(this->mArray), this->index(n, 0), value);
+    AccessorEval<CodecType, ValueType>::set(this->mSetter, const_cast<AttributeArray*>(this->mArray), this->index(n, 0), value);
```

Comment:
I don't think that will help actually, because we're still only storing a single const pointer to the array which is in the AttributeHandle. I'd rather not add another (non-const) one to the AttributeWriteHandle if possible.